### PR TITLE
fix: x,y offset responsive

### LINKED ISSUE #130: Fix  X, Y offset for add labels step within different screens size

### PR DESCRIPTION
**Pull request by LinaYahya**
closes #130

### CODE REVIEW

File: src/modules/context/imageDimensionContext.tsx [hunk 1]
```diff
@@ -1,61 +1,72 @@
...
+};
 
 const ImageDimensionsContext = createContext<ImageDimensionContextType>({
   imgRef: null,
-  saveImageDimension: () => {},
   dimension: { width: 0, height: 0 },
 });
 
-type Props = {
+type Props1 = {
```
Comment (pyphilia) [inline src/modules/context/imageDimensionContext.tsx:23]:
```suggestion
type Props = {
```

File: src/modules/context/imageDimensionContext.tsx [hunk 2]
```diff
@@ -1,61 +1,72 @@
...
   children: JSX.Element;
 };
+export const ImageDimensionsProvider = ({ children }: Props1): JSX.Element => {
```
Comment (pyphilia) [inline src/modules/context/imageDimensionContext.tsx:26]: Add comments why image dimensions are necessary.

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the changes, I think we can even more improve this code, let me know if you have questions!

File: src/modules/common/PlayerFrame.tsx [hunk 3]
```diff
@@ -19,84 +20,94 @@ const PlayerFrame = (): JSX.Element => {
...
+
+      setAnsweredLabels(answered);
+      setLabels(settingLabels);
     }
   }, [appSettings]);
 
+  const findAnLabelIndex = (id: string): number =>
+    answeredLabels?.findIndex(({ expected }) => expected.id === id) ?? -1;
+
   const onDragEnd = (draggable: DropResult): void => {
```
Comment (ReidyT) [inline src/modules/common/PlayerFrame.tsx:46]: `onDragEnd` is a bit long, what do you think about extracting some of its logic in smaller functions ?

**Review by ReidyT — APPROVED** [review]
Thanks for you PR, LGTM 🤠 I just let some suggestions

File: src/utils/dnd.ts [hunk 4]
```diff
@@ -22,3 +23,98 @@ export const updateLabels = (
...
+};
+
+export const trackLabelsChanges = ({
+  srcDroppableId,
+  distDroppableId,
+  labels,
+  answeredLabels,
+  srcLabelIndex,
+}: MoveContext): UpdatedLabels => {
+  const handleMoveFromAll = ({
```
Comment (ReidyT) [inline src/utils/dnd.ts:55]: Maybe I'd finally put these three functions outside the trackLabelsChanges to improve a bit the readability of this function 😅

File: src/utils/dnd.ts [hunk 5]
```diff
@@ -22,3 +23,98 @@ export const updateLabels = (
...
+
+type MoveContext = {
+  labels: Label[];
+  answeredLabels: AnsweredLabel[];
+  srcDroppableId: string;
+  distDroppableId: string;
+  srcLabelIndex: number;
+};
+
+export const trackLabelsChanges = ({
```
Comment (ReidyT) [inline src/utils/dnd.ts:48]: What do you think about this suggestion ?

```suggestion
export const handleMoveLabel = ({
```

**Review by ReidyT — APPROVED** [review]
Thanks for your work and your changes! I just let some suggestions 😆

**Pull request merged**